tests/unit/components/Editor/ItemEditor.spec.js

```diff
@@ -1,347 +1,541 @@
+
+    it("clearing max limit sets it to the minimum value", async () => {
+      await wrapper
+        .find('[data-test="maxCharLimit"]')
+        .find('[data-test="input"]')
+        .setValue("");
+      expect(wrapper.vm.maxCharLimit).toBe(100);
+    });
+
+    it("triggers keyup on max char limit input box", async () => {
+      const maxCharLimitInputKeypress = jest.spyOn(
+        ItemEditor.methods,
+        "maxCharLimitInputKeypress"
+      );
+      mountWrapper();
+      await wrapper
+        .find('[data-test="maxCharLimit"]')
+        .find('[data-test="input"]')
+        .trigger("keypress", {
+          key: 1,
+        });
+      expect(maxCharLimitInputKeypress).toHaveBeenCalled();
+    });
+
+    it("triggers keydown on max char limit input box", () => {
+      const maxCharLimitInputKeydown = jest.spyOn(
+        ItemEditor.methods,
+        "maxCharLimitInputKeydown"
+      );
+      mountWrapper();
+      wrapper
+        .find('[data-test="maxCharLimit"]')
+        .find('[data-test="input"]')
+        .trigger("keydown", {
+          key: 1,
+        });
+      expect(maxCharLimitInputKeydown).toHaveBeenCalled();
     });
-    expect(wrapper.find('[data-test="options"]').exists()).toBeFalsy();
-    expect(
-      wrapper.find('[data-test="subjectiveQuestionContainer"]').exists()
-    ).toBeTruthy();
   });
 
-  it("clicking on question type dropdown toggles the dropdown display", async () => {
-    const wrapper = mount(ItemEditor, {
-      props: {
-        itemList: [
-          {
-            type: "question",
-            time: 13,
-          },
-        ],
-        itemDetailList: [
-          {
-            text: "test",
-            type: "subjective",
-          },
-        ],
-        questionTypeIndex: 1,
-      },
+  describe("buttons", () => {
+    it("clicking add item hides item editor", async () => {
+      const removeSelectedItemIndex = jest.spyOn(
+        ItemEditor.methods,
+        "removeSelectedItemIndex"
+      );
+      mountWrapper();
+      await wrapper.find('[data-test="addItem"]').trigger("click");
+      expect(removeSelectedItemIndex).toHaveBeenCalled();
     });
+
+    it("emits on clicking delete item", async () => {
+      await wrapper.find('[data-test="deleteItem"]').trigger("click");
+      expect(wrapper.emitted()).toHaveProperty("delete-selected-item");
+    });
+
+    describe("multiple items", () => {
+      let updatedItemList = clonedeep(itemList);
+      let updatedItemDetailList = clonedeep(itemDetailList);
+      updatedItemList.push({
+        type: "question",
+        time: 20,
+      });
+      updatedItemDetailList.push({
+        text: "test",
+        type: "mcq",
+        options: ["", ""],
+      });
+
+      it("clicking next item navigates to next item", async () => {
+        const updateSelectedItemIndex = jest.spyOn(
+          ItemEditor.methods,
+          "updateSelectedItemIndex"
+        );
+        mountWrapper(updatedItemList, updatedItemDetailList);
```

Comment:
instead of mounting everything again you can just set the props na using `setProps`?